numpy/random/_generator.pyx

```diff
@@ -3537,10 +3537,9 @@ cdef class Generator:
             # approximately zero or when the covariance is not positive-semidefinite
             _factor = u * np.sqrt(abs(s))
         else:
-            _factor = np.sqrt(s)[:, None] * vh
+            _factor = u * np.sqrt(s)
```

Comment:
The problem here is that `u` is not necessarily `vh.T` for the `svd`, if there are negative eigenvalues from numerical error, there will be a sign change. Hmm... if the cov is valid and the negative eigenvalue is due to numerical error, i.e., it is really tiny, then it should have hardly any impact on the output, so this is probably OK. Might mention the possibility in the release note.

Reply:
I mean the same is true for the previous formulation, right?

Reply:
Yep, it would give incorrect results in that case, but different incorrect results :)

Reply:
ok let's make sure our errors are correct then